### PR TITLE
fix(csp): update ol and fix webpack global emit (#2138)

### PR DESCRIPTION
**Pull request by dagda1**
**Summary**

We are using netlify-cms for a client who has a strict csp policy that would forbid adding:

```
"'unsafe-inline'",
"'unsafe-eval'",
```

to the csp.

After looking into the problem, it would appear that [rbush](https://github.com/mourner/rbush) which is here in the dependency tree:

```
netlify-cms
└─┬ netlify-cms-widget-map@1.5.0 -> ./packages/netlify-cms-widget-map
  └─┬ ol@5.3.3
    └── rbush@2.0.2
```

Is using `new Function("blah"` calls which will get blocked by csp unless the `unsafe-` directives are added.

Upgrading `ol` to `6.9.0` upgrades `rbush` to `3.0.1` where the `new Function` calls have been replaced:

```
netlify-cms
└─┬ netlify-cms-widget-map@1.5.0 -> ./packages/netlify-cms-widget-map
  └─┬ ol@6.9.0
    └── rbush@3.0.1
   
**Checklist**

Please add a `x` inside each checkbox:

- [x] I have read the [contribution guidelines](../CONTRIBUTING.md).
- [x] Code is formatted via running `yarn format`.
- [x] Tests are passing via running `yarn test`.
- [ ] The status checks are successful (continuous integration). Those can be seen below.

### CODE REVIEW

**Review by erezrokah — CHANGES_REQUESTED** [review]
Thanks @dagda1, this breaks our demo site and our e2e tests.

See error from the [deploy preview](https://deploy-preview-6009--cms-demo.netlify.app/):
image

File: packages/netlify-cms-proxy-server/webpack.config.js [hunk 1]
```diff
@@ -10,7 +10,7 @@ module.exports = {
   entry: { index: path.join('src', 'index.ts'), middlewares: path.join('src', 'middlewares.ts') },
   mode: NODE_ENV,
   target: 'node',
-  devtool: 'source-map',
+  devtool: 'cheap-module-source-map',
```
Comment (erezrokah) [inline packages/netlify-cms-proxy-server/webpack.config.js:13]: Hi @dagda1, can you explain the motivation for this change?

Based on https://webpack.js.org/configuration/devtool/#devtool we might want to have the original high quality source maps in our production builds.

File: packages/netlify-cms-widget-map/package.json [hunk 2]
```diff
@@ -30,6 +30,6 @@
     "react-immutable-proptypes": "^2.1.0"
   },
   "dependencies": {
-    "ol": "^5.3.1"
+    "ol": "^6.9.0"
```
Comment (erezrokah) [inline packages/netlify-cms-widget-map/package.json:33]: I think this version has breaking changes and can't be updated as is.
It breaks the map widget, see the deploy preview:
image

https://deploy-preview-6009--cms-demo.netlify.app/#/collections/kitchenSink/entries/a-big-entry-with-all-the-things

Comment (dagda1) [reply packages/netlify-cms-widget-map/package.json:33]: Any suggestions on what to to here?

The reason for creating the PR is because we have a non-negotiable strict CSP policy and there are `new Function` eval code being emitted from netlify-cms and its dependencies as a colleague of mine outline in #2138.

Our other alternative is to use a different CMS and judging by this PR, that might be the easier option.

Fixing this and the ajv schema `new Function` emissions is not something I am relishing.

I don't know the codebase at all.

Comment (dagda1) [reply packages/netlify-cms-widget-map/package.json:33]: I don't see any tests in the `netlify-cms-widget-map` package, should there be?

Comment (dagda1) [reply packages/netlify-cms-widget-map/package.json:33]: I don't see any tests for this widget

Comment (erezrokah) [reply packages/netlify-cms-widget-map/package.json:33]: Hi @dagda1, we don't have automated tests for this widget, but if you try to use the deploy preview (the link I shared), you'd see it's broken.

The breaking changes are documented in https://github.com/openlayers/openlayers/releases/tag/v6.0.0, if you'd like to take a look.

I understand the CSP issue, but we'd like to avoid breaking another feature of the CMS while we handle it

Comment (dagda1) [reply packages/netlify-cms-widget-map/package.json:33]: @erezrokah no tests in the package and no types make this really difficult to someone who is looking at this for the first time.  

I was hoping for at least some collaboration with someone who actually knows this code but instead, I have been given a link.

Not a lot for me to go on here.

Is the bad CSP not important to netlify?

Comment (dagda1) [reply packages/netlify-cms-widget-map/package.json:33]: @erezrokah I have [over 1000 open source contributions](https://github.com/dagda1) in the last year and this is without question the least help I received from one of the biggest companies.

Also, what I am suggesting is of benefit so I am pretty shocked to say the least.

Comment (erezrokah) [reply packages/netlify-cms-widget-map/package.json:33]: Hi @dagda1, sorry for creating this impression.

The CMS is a community project and thus most of the work is expected to come from external contributors.

I understand the benefit of fixing the CSP issue, and the importance of the issue can also be seen based on the upvotes [it receives from the community](https://github.com/netlify/netlify-cms/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc).

I do think that breaking one feature in order to fix another issue is not something that we should introduce.

For the lack of tests and types, this has been an ongoing effort and we've been introducing both when we make a code change to an area that doesn't have them.

The best way to move forward on this would be to look at the map's widget code here:
https://github.com/netlify/netlify-cms/blob/f1e5dbc9ea75570ffe5163cfdad3245699fca378/packages/netlify-cms-widget-map/src/withMapControl.js#L72
and understand the necessary changes required for it to support the major version bump of the `ol` library.

That's the only place we're using the `ol` library so the fix should be scoped to that widget.

Comment (dagda1) [reply packages/netlify-cms-widget-map/package.json:33]: > and understand the necessary changes required for it to support the major version bump of the ol library.

I have found that code some time ago and the reason the widget is not rendering is because  the map container's width or height are 0.

[This warning](https://github.com/openlayers/openlayers/blob/c1f4749e4d4701c73be1c3787a700d0fed8cfd7d/src/ol/PluggableMap.js#L1530) is triggered

Comment (dagda1) [reply packages/netlify-cms-widget-map/package.json:33]: > I do think that breaking one feature in order to fix another issue is not something that we should introduce.

who said anything snout committing broken code.  all i am asking gif and totally not getting is done input from someone who knows the code better than me.





> The best way to move forward on this would be to look at the map's widget code here:
> 
> https://github.com/netlify/netlify-cms/blob/f1e5dbc9ea75570ffe5163cfdad3245699fca378/packages/netlify-cms-widget-map/src/withMapControl.js#L72
> 

that is as i said the least helpful advice i have received in over 1000 commits to open source projects this year.

look at the code is what you are saying.

no sh*t sherlock.  some things don't need to be stayed

Comment (dagda1) [reply packages/netlify-cms-widget-map/package.json:33]: @erezrokah If I give the container div a fixed height then if you ask me, the widget works.

Any idea why the container div would not have a 0 height:

netlify-shit

Comment (erezrokah) [reply packages/netlify-cms-widget-map/package.json:33]: Woo, great find. Let me try and isolate this problem locally.
They did a whole lot of style changes in the new version, so maybe this is just about setting a `min-height`

Comment (dagda1) [reply packages/netlify-cms-widget-map/package.json:33]: > 

Thank you

Comment (erezrokah) [reply packages/netlify-cms-widget-map/package.json:33]: So adding `height: 400px;` after this line https://github.com/netlify/netlify-cms/blob/9f0d11957c139cc274d4ccde0ac537d37af3b9b2/packages/netlify-cms-widget-map/src/withMapControl.js#L82 seems to fix this issue.

I can't seem to push this change as it says I don't have access to the fork, do you mind doing this change?

Comment (dagda1) [reply packages/netlify-cms-widget-map/package.json:33]: @erezrokah I could add 400px but would it not be better to make height a prop with 400px being the default?

Giving everyone 400px sounds less than ideal although I'm happy to do that

Comment (erezrokah) [reply packages/netlify-cms-widget-map/package.json:33]: My guess is that at some point it became the container responsibility to handle this, but I couldn't find the change that introduced it

Comment (erezrokah) [reply packages/netlify-cms-widget-map/package.json:33]: > I could add 400px but would it not be better to make height a prop with 400px being the default?

Even better!

Comment (dagda1) [reply packages/netlify-cms-widget-map/package.json:33]: @erezrokah I've done that in [this commit](https://github.com/netlify/netlify-cms/pull/6009/commits/a1dba53f05b1bb466c0949ff20fcec87a23f9637) and the map is rendering.

I've also reset `devtool` to `source-map`.

Are there any other checks we need to make?

File: babel.config.js [hunk 3]
```diff
@@ -28,12 +27,16 @@ const defaultPlugins = [
 ];
 
 const svgo = {
-  plugins: extendDefaultPlugins([
+  plugins: [
```
Comment (erezrokah) [inline babel.config.js:30]: Is this change related to this PR?

**Review by erezrokah — CHANGES_REQUESTED** [review]

File: packages/netlify-cms-widget-map/package.json (hunk 2 above)
Comment (erezrokah) [reply packages/netlify-cms-widget-map/package.json:33]: Added another comment here https://github.com/netlify/netlify-cms/pull/6009#discussion_r764160640

**Comment by erezrokah** [comment]
Added a comment, also not sure what's up with this check https://github.com/netlify/netlify-cms/pull/6009/checks?check_run_id=4446496928 as we don't use Azure pipelines

**Comment by dagda1** [comment]
> Added a comment, also not sure what's up with this check https://github.com/netlify/netlify-cms/pull/6009/checks?check_run_id=4446496928 as we don't use Azure pipelines

I think it is because there was a merge conflict which is now fixed.

Everything is.........green (at this time of writing).

**Review by erezrokah — APPROVED** [review]
🚀 This is good to go
